### PR TITLE
docs(retry): add example for retry operator

### PR DESCRIPTION
**Pull request by waseemahmad31**
**Description:** Updated documentation for retry operator

**Related issue (if exists):** #4627

### CODE REVIEW

**Comment by waseemahmad31** [comment]
getting build error `expect  Expected type to be:  Observable<{}>got Observable<unknown>`, not sure why getting this error.

**Comment by benlesh** [comment]
@waseemahmad31 it was a problem with dtslint tests. You can rebase and it will be fine now. Sorry about that.

File: src/internal/operators/retry.ts [hunk 1]
```diff
@@ -15,6 +15,36 @@ import { MonoTypeOperatorFunction, TeardownLogic } from '../types';
...
  * would be: [1, 2, 1, 2, 3, 4, 5, `complete`].
+ *
+ * ## Example
+ * ```ts
+ * import { interval, of, throwError } from 'rxjs';
+ * import { flatMap, retry } from 'rxjs/operators';
+ *
+ * const source = interval(1000);
+ * const example = source
+ *   .flatMap(val => {
```
Comment (Edouardbozon) [inline src/internal/operators/retry.ts]: What about using Pipeable operators ?

Comment (waseemahmad31) [reply src/internal/operators/retry.ts]: done.

**Comment by waseemahmad31** [comment]
> @waseemahmad31 it was a problem with dtslint tests. You can rebase and it will be fine now. Sorry about that.

Thanks @benlesh

**Comment by waseemahmad31** [comment]
@benlesh, @Edouardbozon please review this pull request and let me know if any changes needed in order to merge this pull request.

**Pull request merged**